src/extlib/logger.hpp

```diff
@@ -0,0 +1,88 @@
+#ifndef CPPLOGGER_HPP
+#define CPPLOGGER_HPP
+
+#include <chrono>
+#include <iostream>
+#include <string>
+#include <fstream>
+#include <filesystem>
+#include <sstream>
+#include <iomanip>
+
+enum class LogLevel {
+    INFO,
+    WARNING,
+    ERROR,
+    DEBUG
+};
+
+enum class LogOutput {
+    CONSOLE,
+    FILE
+};
+
+class Logger {
+public:
+    static Logger* init_logging() {
+        static Logger instance;
+        return &instance;
+    }
+
+    void log(LogLevel log_level, LogOutput log_output,const std::string& message) {
+        std::string log_message = get_current_timestamp_str() +
+                                  get_log_level_str(log_level) +
+                                  message;
+        log_to_file(log_message);
+
+        if (log_output == LogOutput::CONSOLE) {
+            std::cout << log_message << std::endl;
+        }
+    }
+
+    void set_log_filepath(const std::string& filepath) {
+        _log_filepath = filepath;
+    }
+
+private:
+    std::string _log_filepath;
+
+    std::string get_log_level_str(LogLevel level) const {
+        switch (level) {
+            case LogLevel::ERROR:
+                return "[ERROR]  ";
```

Comment:
`case` should have the same indentation as `switch`
```suggestion
        switch (level) {
        case LogLevel::ERROR:
            return "[ERROR]  ";
```